src/scripting_api/util.js

```diff
@@ -20,11 +20,724 @@ class Util extends PDFObject {
+            break;
+          }
+        }
+        intPart = buf.reverse().join(thousandSep);
+      } else {
+        intPart = intPart.toString();
+      }
+
+      let n = `${intPart}${decPart}`;
+      if (nWidth !== undefined) {
+        n = n.padStart(nWidth - prefix.length, cFlags & ZERO ? "0" : " ");
+      }
+
+      return `${prefix}${n}`;
+    });
+  }
+
+  iconStreamFromIcon() {
+    /* Not implemented */
+  }
+
+  printd(cFormat, cDate, bXFAPicture = false) {
+    if (bXFAPicture) {
+      return this._printdXFAPicture(cFormat, cDate);
+    }
+
+    switch (cFormat) {
+      case "0":
+        return this.printd("D:yyyymmddHHMMss", cDate);
+      case "1":
+        return this.printd("yyyy.mm.dd HH:MM:ss", cDate);
+      case "2":
+        return this.printd("m/d/yy h:MM:ss tt", cDate);
+    }
+
+    const handlers = {
+      mmmm: data => {
+        return this._months[data.month];
+      },
+      mmm: data => {
+        return this._months[data.month].substring(0, 3);
+      },
+      mm: data => {
+        return (data.month + 1).toString().padStart(2, "0");
+      },
+      m: data => {
+        return (data.month + 1).toString();
+      },
+      dddd: data => {
+        return this._days[data.dayOfWeek];
+      },
+      ddd: data => {
+        return this._days[data.dayOfWeek].substring(0, 3);
+      },
+      dd: data => {
+        return data.day.toString().padStart(2, "0");
+      },
+      d: data => {
+        return data.day.toString();
+      },
+      yyyy: data => {
+        return data.year.toString();
+      },
+      yy: data => {
+        return (data.year % 100).toString().padStart(2, "0");
+      },
+      HH: data => {
+        return data.hours.toString().padStart(2, "0");
+      },
+      H: data => {
+        return data.hours.toString();
+      },
+      hh: data => {
+        return (1 + ((data.hours + 11) % 12)).toString().padStart(2, "0");
+      },
+      h: data => {
+        return (1 + ((data.hours + 11) % 12)).toString();
+      },
+      MM: data => {
+        return data.minutes.toString().padStart(2, "0");
+      },
+      M: data => {
+        return data.minutes.toString();
+      },
+      ss: data => {
+        return data.seconds.toString().padStart(2, "0");
+      },
+      s: data => {
+        return data.seconds.toString();
+      },
+      tt: data => {
+        return data.hours >= 1 && data.hours <= 12 ? "am" : "pm";
+      },
+      t: data => {
+        return data.hours >= 1 && data.hours <= 12 ? "a" : "p";
+      },
+    };
+
+    const year = cDate.getFullYear();
+    const month = cDate.getMonth();
+    const day = cDate.getDate();
+    const dayOfWeek = cDate.getDay();
+    const hours = cDate.getHours();
+    const minutes = cDate.getMinutes();
+    const seconds = cDate.getSeconds();
+    const data = { year, month, day, dayOfWeek, hours, minutes, seconds };
+
+    const pattern = /(mmmm|mmm|mm|m|dddd|ddd|dd|d|yyyy|yy|HH|H|hh|h|MM|M|ss|s|tt|t|\\.)/g;
```

Comment:
Why is the `\\.` part here?

Reply:
see https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/js_api_reference.pdf#page=719
...
`\` use as an escape character